pkg/dockerregistry/server/signaturedispatcher.go

```diff
@@ -0,0 +1,147 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+
+	ctxu "github.com/docker/distribution/context"
+
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/registry/api/errcode"
+	"github.com/docker/distribution/registry/handlers"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+
+	gorillahandlers "github.com/gorilla/handlers"
+)
+
+type Signature struct {
+	Name    string `json:"name"`
+	Type    string `json:"type"`
+	Content []byte `json:"content"`
+}
+
+type SignatureList struct {
+	Signatures []Signature `json:"signatures"`
+}
+
+const defaultSignatureType = "atomic"
```

Comment:
Could we perhaps use predefined [ImageSignatureTypeAtomicImageV1](https://github.com/miminar/origin/blob/existing-images-pushed-with-correct-ref/pkg/image/api/types.go#L93) instead?